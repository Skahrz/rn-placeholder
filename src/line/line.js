import React from "react";
import { View } from "react-native";

export const Line = ({
  textSize = 12,
  color = "#efefef",
  style,
  noMargin = false,
  ...props
}) => {
  const height = textSize;
  const alignSelf = "stretch";
  const backgroundColor = color;
  const borderRadius = textSize / 4;
  const marginBottom = noMargin ? 0 : textSize;
  const computedStyle = {
    height,
    alignSelf,
    backgroundColor,
    borderRadius,
    marginBottom,
    marginHorizontal: 0
  };

  return <View style={[computedStyle, style]} {...props} />;
};
