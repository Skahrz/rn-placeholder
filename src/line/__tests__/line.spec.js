import React from "react";
import { shallow } from "enzyme";
import { Line } from "../line";

describe("Line", () => {
  let props;

  const getWrapper = () => shallow(<Line {...props} />);

  beforeEach(() => {
    props = {};
  });

  it("should match snapshot", () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it("should match snapshot with a bit more props and style", () => {
    props.style = { marginTop: 1000 };
    props.testID = "123";
    props.noMargin = true;

    expect(getWrapper()).toMatchSnapshot();
  });
});
