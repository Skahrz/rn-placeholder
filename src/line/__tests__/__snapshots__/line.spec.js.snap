// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#efefef",
        "borderRadius": 3,
        "height": 12,
        "marginBottom": 12,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;

exports[`Line should match snapshot with a bit more props and style 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#efefef",
        "borderRadius": 3,
        "height": 12,
        "marginBottom": 0,
        "width": "100%",
      },
      Object {
        "marginTop": 1000,
      },
    ]
  }
  testID="123"
/>
`;
